### ISSUE TITLE
Two questions about image classification distributed training.

### ISSUE THREAD

**Issue by TheodoreG** [PaddlePaddle/Paddle#14377]
I have used the distributed training code for image classification in [https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleCV/image_classification/dist_train/dist_train.py](url) and got some questions.
1. It seems that variable nccl_id_var keeps None in the whole program, and doesn't change even after the append_nccl2_prepare() function is called. Buy the way, function append_nccl2_prepare() seems returned nothing. 
2. In the training loop I can't tell how the program allocate different training data batch into different device. Ideally, different devices should train different training batches, or may be ParallelExecutor do such things for me ?

**Comment by typhoonzero** [comment]
Thanks for reporting. These two are very good questions.

1. for nccl2 mode, I've updated `append_nccl2_prepare` to use latest transpiler API to simplify the code. Actually, you should not care about `nccl_id_var`, will remove this in next commit.
2. data partition is done by the reader: https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleCV/image_classification/reader.py#L139. Yet there's still issue for the current shuffling method, each node should use one part of the shuffled whole data set for each epoch, I'll update this in next commit too.

**Comment by TheodoreG** [comment]
Ok, thanks a lot for your nice explanation.
